### PR TITLE
New GraphQL API

### PR DESCRIPTION
**Pull request by cartogram**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+  auth0Token         String         @unique
+  auth0Id            String         @unique
+  emailSubscriptions Boolean        @default(true)
+  fondfolios         Fondfolio[]
+  contributions      Contribution[]
+  orders             Order[]
 }
 
 model Fondfolio {
+  id                    String         @default(uuid()) @id
```
Comment (cartogram) [inline prisma/schema.prisma:27]:
```suggestion
  id                    String         @default(uuid()) @id
  covers            
```

File: prisma/schema.prisma [hunk 2]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
-  contributions Contribution[]
+  id                 String         @default(uuid()) @id
+  createdAt          DateTime       @default(now())
+  updatedAt          DateTime       @default(now())
+  role               Role           @default(USER)
+  email              String         @unique
+  name               String
+  picture            String
+  auth0Token         String         @unique
+  auth0Id            String         @unique
```
Comment (feeohnah) [inline prisma/schema.prisma:19]: In our old schema this was `auth0UserId`

File: prisma/schema.prisma [hunk 3]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
-  id            String         @default(uuid()) @id
-  role          Role           @default(USER)
-  email         String         @unique
-  password      String
-  fondfolios    Fondfolio[]
-  contributions Contribution[]
+  id                 String         @default(uuid()) @id
+  createdAt          DateTime       @default(now())
+  updatedAt          DateTime       @default(now())
+  role               Role           @default(USER)
```
Comment (feeohnah) [inline prisma/schema.prisma:14]: This was plural `roles` in our old schema

File: prisma/schema.prisma [hunk 4]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+  email              String         @unique
+  name               String
+  picture            String
+  auth0Token         String         @unique
+  auth0Id            String         @unique
+  emailSubscriptions Boolean        @default(true)
+  fondfolios         Fondfolio[]
+  contributions      Contribution[]
+  orders             Order[]
```
Comment (feeohnah) [inline prisma/schema.prisma:23]:
```suggestion
  orders             Order[]
  stripeId: String
```

Comment (feeohnah) [inline prisma/schema.prisma:23]: In our old schema this was ` purchases: [Purchase!]! @relation(name: "UserPurchases")`

File: prisma/schema.prisma [hunk 5]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+  MEET
+  MEMORIALIZED
+  MEMORY
+  MISS
+  NEWPLACE
+  PARENTHOOD
+  WISDOM
+  WRINKLY
+  ADMIRED
+}
```
Comment (feeohnah) [inline prisma/schema.prisma:116]:
```suggestion
}
enum Cover {
MAPLE
CEDAR
WALNUT
}
```

File: prisma/schema.prisma [hunk 6]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+  subtitle              String
+  credit                String?
   shortQuestionnaireUrl String?
-  title                 String?
+  title                 String
   websiteUrl            String?
+  state                 State          @default(COLLECTING)
   contributions         Contribution[]
+  emailSubscriptions    Boolean        @default(true)
+  language              Language       @default(EN)
```
Comment (feeohnah) [inline prisma/schema.prisma:45]:
```suggestion
  language              Language       @default(EN)
  cover Cover @default(MAPLE)
```

File: prisma/schema.prisma [hunk 7]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
   userId                String
-  id                    String         @default(uuid()) @id
-  reciever              String?
+  event                 Event
+  receiver              String[]
+  postface              String
+  preface               String
+  subtitle              String
+  credit                String?
   shortQuestionnaireUrl String?
```
Comment (feeohnah) [inline prisma/schema.prisma:39]: this was just `shortUrl` on the old schema

File: prisma/schema.prisma [hunk 8]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
   user                  User           @relation(fields: [userId], references: [id])
   userId                String
-  id                    String         @default(uuid()) @id
-  reciever              String?
+  event                 Event
+  receiver              String[]
+  postface              String
+  preface               String
+  subtitle              String
+  credit                String?
```
Comment (feeohnah) [inline prisma/schema.prisma:38]: we just had `shortUrl` in the old schema

File: prisma/schema.prisma [hunk 9]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
   shortQuestionnaireUrl String?
-  title                 String?
+  title                 String
   websiteUrl            String?
+  state                 State          @default(COLLECTING)
   contributions         Contribution[]
+  emailSubscriptions    Boolean        @default(true)
+  language              Language       @default(EN)
+  orders                Order[]
+  questions             Question[]
```
Comment (feeohnah) [inline prisma/schema.prisma:47]: We also had `public: Boolean @defaultValue(value: false)
` in the old schema but I'm not sure what that was for?

File: prisma/schema.prisma [hunk 10]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+  id                    String         @default(uuid()) @id
+  createdAt             DateTime       @default(now())
+  updatedAt             DateTime       @default(now())
+  date                  DateTime       @default(now())
   user                  User           @relation(fields: [userId], references: [id])
   userId                String
-  id                    String         @default(uuid()) @id
-  reciever              String?
+  event                 Event
+  receiver              String[]
```
Comment (feeohnah) [inline prisma/schema.prisma:34]: We also have `giver: String` in the old schema, useful for when it's from multiple people using one account

File: prisma/schema.prisma [hunk 11]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+enum State {
+  COLLECTING
+  COMPLETED
+}
+
+enum Event {
+  BIRTHDAY
+  ANNIVERSARY
+  WEDDING
+  REMEBERING
```
Comment (feeohnah) [inline prisma/schema.prisma:93]:
```suggestion
  REMEMBERING
  ENGAGEMENT
  CASE
  MOVING
  RECOVERY
  NEWBORN
  RETIREMENT
  SUCCESS
  GRADUATION
```

File: prisma/schema.prisma [hunk 12]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+}
+
 enum Role {
   USER
   ADMIN
 }
+
+enum State {
+  COLLECTING
+  COMPLETED
```
Comment (feeohnah) [inline prisma/schema.prisma:86]: We also had `PURCHASED` in the old schema

File: prisma/schema.prisma [hunk 13]
```diff
@@ -8,35 +8,109 @@ datasource db {
...
+enum Question {
+  ADMIRE
+  ADVICE
+  ANYTHING
+  BOND
+  COMFORT
+  GRAPEVINE
+  INANIMATE
+  INGREDIENT
+  INTHREE
```
Comment (feeohnah) [inline prisma/schema.prisma:105]: this should be `INSEVEN` or `SEVENWORDS` cause it's the 7 word bio question, it used to be a 3 descriptors questions but we changed it

File: prisma/migrations/20200610050740-fdas/schema.prisma [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+  orders                Order[]
+  questions             Question[]
+}
+
+model Contribution {
+  id          String    @default(uuid()) @id
+  createdAt   DateTime  @default(now())
+  updatedAt   DateTime  @default(now())
+  author      String
+  fondfolio   Fondfolio @relation(fields: [fondfolioId], references: [id])
```
Comment (feeohnah) [inline prisma/migrations/20200610050740-fdas/schema.prisma:55]: Our old schema also had `answers: [Answer!]! @relation(name: "ContributionAnswers")`

File: prisma/migrations/20200610050740-fdas/schema.prisma [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+  fondfolioId String
+  user        User      @relation(fields: [userId], references: [id])
+  userId      String
+}
+
+model Order {
```
Comment (feeohnah) [inline prisma/migrations/20200610050740-fdas/schema.prisma:61]: This is the Purchase model from the old schema `type Purchase @model {
  amount: Int
  createdAt: DateTime!
  description: String
  id: ID! @isUnique
  isPaid: Boolean @defaultValue(value: false)
  purchaseItems: [PurchaseItem!]! @relation(name: "PurchaseOnPurchaseItem")
  stripeChargeId: String
  stripeToken: String
  updatedAt: DateTime!
  user: User @relation(name: "UserPurchases")
}`

Comment (cartogram) [reply prisma/migrations/20200610050740-fdas/schema.prisma:61]: you can do multiline code blocks with 3 backticks

**Pull request merged**